src/scipp/operations.py

```diff
@@ -65,11 +67,18 @@ def elemwise_func(func: Callable,
     Note that ``fmadd(x, y, z)`` would have the same effect in this case, but requires
     a potentially large intermediate allocation for the result of "a * b".
     """
-    func = _as_numba_cfunc(func, unit_func=unit_func)
 
-    def transform_custom(*args: Variable) -> Variable:
-        if auto_convert_dtypes:
-            args = [arg.to(dtype='float64', copy=False) for arg in args]
-        return cpp_transform(func, *args)
+    def decorator(f):
+        f = _as_numba_cfunc(f, unit_func=unit_func)
 
-    return transform_custom
+        @functools.wraps(f)
```

Comment:
This `f` should refer to the input to `decorator` and not the return value of `as_numba_func`